### PR TITLE
feat: [FX-4053] move rich text editor to a separate package

### PR DESCRIPTION
**Pull request by dmaklygin**
[FX-4053]

### Description

We should create a new package `@toptal/picasso-rich-text-editor`.

### How to test

- Use temploy to check workability

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Self reviewed
- [x] Covered with tests

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4053]: https://toptal-core.atlassian.net/browse/FX-4053?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dmaklygin** [comment]
@toptal-anvil ping reviewers

**Review by TomasSlama — APPROVED** [review]
Tested in the temploy, looks good

File: .storybook/main.js [hunk 1]
```diff
@@ -150,6 +150,14 @@ module.exports = {
...
             '../packages/picasso-pictograms/src'
           ),
+          '@toptal/picasso-rich-text-editor': path.resolve(
+            __dirname,
+            '../packages/picasso-rich-text-editor/src'
+          ),
+          '@toptal/picasso-rich-text-editor/utils': path.resolve(
+            __dirname,
+            '../packages/picasso-rich-text-editor/src/utils'
+          ),
```
Comment (augustobmoura) [inline .storybook/main.js:160]: Any special reason we need this alias? I was thinking maybe we can re-export utils from the main package and imports should be something like:
```js
import { utils } from '@toptal/picasso-rich-text-editor'
```

File: packages/picasso-rich-text-editor/README.md
Comment (augustobmoura) [inline packages/picasso-rich-text-editor/README.md:1]: Should we create a README for this package?

File: packages/picasso-rich-text-editor/package.json [hunk 2]
```diff
@@ -0,0 +1,53 @@
+{
+  "name": "@toptal/picasso-rich-text-editor",
+  "version": "1.0.0",
+  "description": "Picasso rich text editor",
+  "author": "Toptal",
+  "homepage": "https://github.com/toptal/picasso/tree/master/packages/picasso-forms#readme",
```
Comment (augustobmoura) [inline packages/picasso-rich-text-editor/package.json:6]: This URL needs to be updated

**Review by augustobmoura — APPROVED** [review]
I left some small comments, but because we are only moving code around it should work regardless

File: .storybook/main.js (hunk 1 above)
Comment (dmaklygin) [reply .storybook/main.js:160]: it's exactly how picasso exports `utils`, I wanted to keep consistency

Comment (augustobmoura) [reply .storybook/main.js:160]: The problem is providing internal paths for imports. Some day, we want to provide a "exports" field on picasso for hiding internal imports, and having a single entrypoint to the package would be great.

But I guess that can wait, Picasso is already working this way, we can change both in the future. Thanks for answering the question!

**Pull request merged**